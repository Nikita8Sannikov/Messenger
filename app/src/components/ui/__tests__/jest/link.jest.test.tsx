import { render,screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import LinkComponent from "@components/ui/link";

describe("LinkComponent", () => {
    it("LinkComponent shows correct", () => {
        render(<LinkComponent>Link</LinkComponent>);
        const linkElement = screen.getByTestId("link");
        expect(linkElement).toBeInTheDocument();
    });

    test("LinkComponent click with userEvent", async () => {
        const user = userEvent.setup();
        const onClick = jest.fn();
        render(<LinkComponent onClick={onClick}>Link</LinkComponent>);
        const linkElement = screen.getByTestId("link");
        await user.click(linkElement);
        expect(onClick).toHaveBeenCalled();
    });

    test("matches snapshot", () => {
        const { asFragment } = render(<LinkComponent></LinkComponent>);

        // Снимаем снапшот компонента
        expect(asFragment()).toMatchSnapshot();
    });
});