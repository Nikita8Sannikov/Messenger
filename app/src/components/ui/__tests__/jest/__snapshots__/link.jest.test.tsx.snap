// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkComponent matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1orsvah-MuiTypography-root-MuiLink-root"
    data-testid="link"
  />
</DocumentFragment>
`;
